src/lib/github-enterprise/utils/index.ts

```diff
@@ -0,0 +1,81 @@
+import * as debugLib from 'debug';
+import Bottleneck from 'bottleneck';
+import fetch from 'node-fetch';
+const debug = debugLib('snyk:github-count');
+
+const limiter = new Bottleneck({
+  maxConcurrent: 1,
+  minTime: 333,
+});
+
+limiter.on('failed', async (error, jobInfo) => {
+  const id = jobInfo.options.id;
+  console.warn(`Job ${id} failed: ${error}`);
+  if (jobInfo.retryCount === 0) {
+    // Here we only retry once
+    console.log(`Retrying job ${id} in 25ms!`);
+    return 25;
+  }
+});
+
+export const fetchAllPages = async (
+  url: string,
+  token: string,
+  itemName?: string,
+): Promise<unknown[]> => {
+  let isLastPage = false;
+  let values: string[] = [];
+  let pageCount = 1;
+  while (!isLastPage) {
+    debug(`Fetching page ${pageCount} for ${itemName}\n`);
+    let response = await limiter.schedule(() =>
+      fetch(url, {
+        method: 'GET',
+        headers: { Authorization: 'Bearer ' + token },
+      }),
+    );
+    if (!response.ok) {
+      if (response.status == 429) {
+        debug(
+          `Failed to fetch page: ${url}\nResponse Status: ${response.status}\nToo many requests \nWaiting for 3 minutes before resuming`,
+        );
+        await sleepNow(180000);
+        debug(`Retrying to fetch page: ${url}`);
+        debug(`Fetching page ${pageCount}\n`);
+        response = await limiter.schedule(() =>
+          fetch(url, {
+            method: 'GET',
+            headers: { Authorization: 'Bearer ' + token },
+          }),
+        );
+      } else if (response.status == 409) {
+        return [];
+      } else {
+        debug(
+          `Failed to fetch page: ${url}\nResponse Status: ${response.status}\nResponse Status Text: ${response.statusText} `,
+        );
+      }
+    }
+    const apiResponse = await response.json();
+    values = values.concat(apiResponse) as string[];
+    const nextPage = response.headers.get('link');
+    if (nextPage && nextPage?.includes('rel="next"')) {
+      const nextPageLink = nextPage
+        .split(',')
+        .find((element) => element.includes('rel="next"'));
+      if (nextPageLink) {
+        url = nextPageLink.split(';')[0].replace('<', '').replace('>', '');
+        if (!url.includes('https')) {
+          url = url.replace('http', 'https');
```

Comment:
was there a situation were the link was http?

Reply:
yes, in the next page url (from the header), it returns "http".

Reply:
done